src/m365/aad/commands/app/app-role-delete.ts

```diff
@@ -0,0 +1,265 @@
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.appObjectId = typeof args.options.appObjectId !== 'undefined';
+    telemetryProps.appName = typeof args.options.appName !== 'undefined';
+    telemetryProps.claim = typeof args.options.claim !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+
+    const deleteAppRole: () => void = (): void => { 
+      this.processAppRoleDelete(logger,args).then(_ => cb(), (rawRes: any): void => this.handleRejectedODataJsonPromise(rawRes, logger, cb));
+    };
+
+    if (args.options.confirm) {
+      deleteAppRole();
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to delete the App role ?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          deleteAppRole();
+        }
+      });
+    }
+  }
+
+  private processAppRoleDelete(logger: Logger, args: CommandArgs):Promise<void>{
+    return this
+      .getAppObjectId(args, logger)
+      .then((appId: string) => this.getAADApp(appId, logger))
+      .then((aadApp: Application): Promise<any> => {
+        
+        const appRoleDeleteIdentifierNameValue = args.options.name ? `name '${args.options.name}'` : (args.options.claim ? `claim '${args.options.claim}'`:`id '${args.options.id}'`) ;
+        if (this.verbose) {
+          logger.logToStderr(`Deleting role with ${appRoleDeleteIdentifierNameValue} from Azure AD app ${aadApp.id}...`);
+        }
+
+        // Find the role search criteria provided by the user.     
+        const appRoleDeleteIdentifierProperty = args.options.name ? `displayName` : (args.options.claim ? `value`:`id`) ;
+        const appRoleDeleteIdentifierValue = args.options.name ? args.options.name : (args.options.claim ? args.options.claim:args.options.id) ;
+      
+        const appRoleToDelete:AppRole[] = aadApp.appRoles!.filter((role:AppRole) => role[appRoleDeleteIdentifierProperty] === appRoleDeleteIdentifierValue);
+
+        if(args.options.name && 
+          appRoleToDelete !== undefined && 
+          appRoleToDelete.length > 1){
+          return Promise.reject(`Multiple roles with the provided 'name' were found. Please disambiguate using the claims : ${appRoleToDelete.map(role => `${role.value}`).join(', ')}`);
+          
+        }
+        if(appRoleToDelete.length === 0){
+          return Promise.reject(`No app role with ${appRoleDeleteIdentifierNameValue} found.`);
+        }
+          
+        const roleToDelete:AppRole = appRoleToDelete[0];
+        
+        if(roleToDelete.isEnabled){
+          return this.disableAppRole(logger,aadApp,roleToDelete.id!).then(() =>{
```

Comment:
We don't need to wrap calling `deleteAppRole` in a function. We could simplify it by creating a promise chain like:

```ts
return this
  .disableAppRole(logger,aadApp,roleToDelete.id!)
  .then(_ => this.deleteAppRole(logger,aadApp,roleToDelete.id!));
```

Reply:
Understood. Fixed